astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -87,6 +94,15 @@ def _to_tree_base_transform_members(cls, model, node, ctx):
         node['fixed'] = dict(model.fixed)
         node['bounds'] = dict(model.bounds)
 
+        if model.input_units is not None:
```

Comment:
I think we only need to serialize the equivalencies on the child models. When the Compound model is constructed back from file, it will recreate its equivalencies from the constituent models. Add another condition here:
`if not isinstance(model, CompoundModel) and model.input_units is not None:` and see if it all works.